### PR TITLE
Raw-tx generation + liquidity manager process changes + withdrawal adaption

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: src/shared/shared.module.ts [hunk 1]
```diff
@@ -27,7 +29,26 @@ import { ApiKeyStrategy } from './auth/api-key.strategy';
...
   controllers: [],
-  providers: [HttpService, JwtStrategy, ApiKeyStrategy, CryptoService, SettingService, AssetService],
-  exports: [PassportModule, JwtModule, ScheduleModule, HttpService, CryptoService, SettingService, AssetService],
+  providers: [
+    HttpService,
+    JwtStrategy,
+    ApiKeyStrategy,
+    CryptoService,
+    JellyfishService,
+    WhaleService,
```
Comment (davidleomay) [inline src/shared/shared.module.ts:38]: Services should only be provided once (Jellyfish and Wale are already in ain.module, Crypto is already in blockchain.module)

File: src/subdomains/staking/staking.module.ts [hunk 2]
```diff
@@ -31,6 +32,7 @@ import { StakingKycCheckService } from './infrastructure/staking-kyc-check.servi
     SharedModule,
     UserModule,
     PayInModule,
+    TransactionModule,
```
Comment (davidleomay) [inline src/subdomains/staking/staking.module.ts:35]: TransactionModule is part of IntegrationModule. Should we rather import Integration here?

File: src/integration/transaction/transaction.module.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Module } from '@nestjs/common';
+import { AinModule } from 'src/blockchain/ain/ain.module';
+import { SharedModule } from 'src/shared/shared.module';
+import { TransactionController } from './api/controllers/transaction.controller';
+import { TransactionExecutionService } from './application/services/transaction-execution.service';
+import { TransactionService } from './application/services/transaction.service';
+
+@Module({
+  imports: [SharedModule, AinModule],
```
Comment (davidleomay) [inline src/integration/transaction/transaction.module.ts:9]: Import BlockchainModule here?

File: src/integration/masternode/domain/entities/masternode.entity.ts [hunk 4]
```diff
@@ -20,8 +20,8 @@ export class Masternode extends IEntity {
   @Column({ nullable: true })
   ownerWallet: string;
 
-  @Column({ type: 'integer', nullable: true })
-  timeLock: number;
+  @Column({ default: MasternodeTimeLock.NONE, nullable: false })
+  timeLock: MasternodeTimeLock;
```
Comment (davidleomay) [inline src/integration/masternode/domain/entities/masternode.entity.ts:24]: Do we really want to have a default time lock value?

Comment (Krysh90) [reply src/integration/masternode/domain/entities/masternode.entity.ts:24]: either we have a default value in the database, or a default value we can fallback to on masternode creation. I have thought that having a default value in the database is more obvious to changing it

File: src/integration/transaction/transaction.module.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Module } from '@nestjs/common';
+import { AinModule } from 'src/blockchain/ain/ain.module';
+import { BlockchainModule } from 'src/blockchain/blockchain.module';
+import { SharedModule } from 'src/shared/shared.module';
+import { TransactionController } from './api/controllers/transaction.controller';
+import { TransactionExecutionService } from './application/services/transaction-execution.service';
+import { TransactionService } from './application/services/transaction.service';
+
+@Module({
+  imports: [SharedModule, AinModule, BlockchainModule],
```
Comment (davidleomay) [inline src/integration/transaction/transaction.module.ts:10]: BlockchainModule should be sufficient (no AinModule)

File: src/subdomains/user/user.module.ts [hunk 6]
```diff
@@ -29,6 +30,7 @@ import { AdminController } from './api/controllers/admin.controller';
       WalletRepository,
     ]),
     SharedModule,
+    AinModule,
```
Comment (davidleomay) [inline src/subdomains/user/user.module.ts:33]: BlockchainModule?

File: src/subdomains/staking/staking.module.ts (hunk 2 above)
Comment (Krysh90) [reply src/subdomains/staking/staking.module.ts:35]: fixed in #42

File: src/integration/masternode/domain/entities/masternode.entity.ts (hunk 4 above)
Comment (Krysh90) [reply src/integration/masternode/domain/entities/masternode.entity.ts:24]: fixed in #42

File: src/integration/transaction/transaction.module.ts (hunk 5 above)
Comment (Krysh90) [reply src/integration/transaction/transaction.module.ts:10]: fixed in #42

File: src/subdomains/user/user.module.ts (hunk 6 above)
Comment (Krysh90) [reply src/subdomains/user/user.module.ts:33]: fixed in #42

**Review by davidleomay — APPROVED** [review]

**Pull request merged**